sympy/ntheory/ec.py

```diff
@@ -32,28 +33,29 @@ class EllipticCurve():
 
     """
 
-    def __init__(self, a4, a6, a1=0, a2=0, a3=0, domain=QQ):
-        self._domain = domain
+    def __init__(self, a4, a6, a1=0, a2=0, a3=0, modulus = 0):
+        if modulus == 0:
+            self._domain = QQ
```

Comment:
I would write `domain = QQ` or `domain = FF(modulus)`, and then `self._domain = domain`. Then those lines below where coefficients are converted: `self._a1 = self._domain(a1)` could be written `self._a1 = domain(a1)`, ...